jwt/jwk.py

```diff
@@ -279,25 +276,139 @@ def jwk_from_dict(dct: Mapping[str, Any]) -> AbstractJWKBase:
     return supported[kty].from_dict(dct)
 
 
-def jwk_from_pem(pem_content: bytes) -> AbstractJWKBase:
-    if not isinstance(pem_content, bytes):  # pragma: no cover
-        raise TypeError('pem_content must be a bytes')
-
+def jwk_from_bytes_argument_conversion(func):
+    if 'private' in func.__name__:
+        kind = 'private'
+    elif 'public' in func.__name__:
+        kind = 'public'
+    else:
+        raise Exception("the wrapped function must have either public"
+                        " or private in it's name")
+
+    def wrapper(content, **kwargs):
```

Comment:
decorator should be decorated with [functools.wraps](https://docs.python.org/3/library/functools.html#functools.wraps) to keep original name and docstring.

Reply:
decorating the decorator with wraps is not mandatory this time: at "compile time" it will raise the exception if the decorated function doesn't have either public or private in their name
anyway it's also good to get the original name and doctring